client/lib/naive-client-side-rollout/test/naive-client-side-rollout.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * Internal dependencies
+ */
+import user from 'calypso/lib/user';
+import badNaiveClientSideRollout from '../naive-client-side-rollout';
+
+jest.mock( 'calypso/lib/user' );
+const mockedUser = user as jest.MockedFunction< typeof user >;
+const setUserId = ( userId ) =>
+	mockedUser.mockImplementationOnce( ( ( () => ( {
+		get: () => ( { ID: userId } ),
+	} ) ) as unknown ) as typeof user );
+
+// [a, b)
+const range = ( a, b ) => {
+	const arr = [];
+	for ( let i = a; i < b; i++ ) {
+		arr.push( i );
+	}
+	return arr;
+};
+
+const numUserIds = 10000;
+const userIds = range( 1, numUserIds + 1 );
+const badNaiveClientSideRolloutWithMockedUserId = ( percent, userId ) => {
+	setUserId( userId );
+	return badNaiveClientSideRollout( 'feature-id', percent );
+};
+
+const numRandomNums = 1000;
+const randomNums = range( 0, numRandomNums ).map( ( x ) => x / numRandomNums );
```

Comment:
This name is a bit confusing. When I first read it I expected the numbers to be deterministically random, not sequential. I think that including the word `mock` would make it clearer.

```suggestion
const numRandomMocks = 1000;
const randomMocks = range( 0, numRandomMocks ).map( ( x ) => x / numRandomMocks );
```

```suggestion
const numRandomNumMocks = 1000;
const randomNumMocks = range( 0, numRandomNumMocks ).map( ( x ) => x / numRandomNumMocks );
```